### PR TITLE
Implement Softplus Function and Tests

### LINKED ISSUE #19296: softplus

### PR DESCRIPTION
**Pull request by BouchanaHicham**
Requesting Team Decision on Softplus Functions

- Created my own Softplus functions and corresponding tests
- Similar Softplus functions were found and commented out
- Both implementations have been thoroughly tested and passed successfully

To proceed, the team needs to decide which implementation of Softplus functions to use:
1. The existing commented-out functions
2. My newly created functions

Please review both implementations and provide feedback on which option to proceed with. Additionally, if there are any changes or additions required in either implementation, please specify them for further refinement.

- Close #19296

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by sagewhocodes** [comment]
Hello @BouchanaHicham , Thank you for contributing to ivy!

However as I can see, the `softplus` function you have reimplemented doesn't have the necessary arguments from the original softmax function from Paddle's API [here](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/nn/Softplus_en.html) . 
Since frontend functions are used for transpiling across frameworks, You are required to ensure function signatures of the frontend function are exactly the same. You can refer to Ivy frontends doc [here](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html)

Happy Contributing 😄

**Comment by BouchanaHicham** [comment]
Hello, @sagewhocodes!

I would greatly appreciate it if you could review my latest changes. I value your expertise and insights, and I'm eager to hear your thoughts on the direction I should take from here.

Thank you in advance for your time and feedback!

Best regards,

**Comment by sagewhocodes** [comment]
Hello @BouchanaHicham , there seems to be redundant changes in `test_activation.py` file. Could you please look into it?

**Comment by BouchanaHicham** [comment]
> Hello @BouchanaHicham , there seems to be redundant changes in `test_activation.py` file. Could you please look into it?

Hello @sagewhocodes ,

Thank you for your feedback. I believe you are referring to the removal of the softplus_test in test_activation.py. Allow me to clarify the rationale behind this change and provide further context.

The previous implementation of the softplus_test had some limitations and inconsistencies. Therefore, I made the decision to remove it and reimplement it to ensure better coverage and accuracy. The updated test now includes the necessary arguments and incorporates improvements to enhance its functionality.

By removing the redundant softplus_test and replacing it with an improved version, i aim to streamline the codebase and maintain consistency throughout the test suite.

I appreciate your attention to detail and your commitment to code quality. If you have any additional questions or concerns, please don't hesitate to let me know.

File: ivy/functional/frontends/paddle/nn/functional/activation.py [hunk 1]
```diff
@@ -231,3 +225,9 @@ def log_sigmoid(x, name=None):
 
 def silu(x, name=None):
     return ivy.silu(x)
+
+
+@with_supported_dtypes({"2.5.0 and below": ("float32", "float64")}, "paddle")
```
Comment (sagewhocodes) [inline ivy/functional/frontends/paddle/nn/functional/activation.py:230]: Revert the changes in this file, as the implementation is unchanged

**Review by sagewhocodes — CHANGES_REQUESTED** [review]

**Comment by sagewhocodes** [comment]
@BouchanaHicham please do the requested changes, rest looks good to me!

**Comment by BouchanaHicham** [comment]
@sagewhocodes Done

**Comment by BouchanaHicham** [comment]
> @BouchanaHicham please do the requested changes, rest looks good to me!

Done

**Comment by sagewhocodes** [comment]
LGTM!

**Pull request merged**